src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2383 @@
+				CModeSolutionSpace solutionspace = Mode.sol;
+				bool SAHR_OC_MetinMode = false;
+				bool SAT_OC_MetinMode = false;
+				int solution_map_sizeX = solutionspace.PointX.size() - 1;
+				int solution_map_sizeY = solutionspace.PointY.size() - 1;
+				
+				if (solution_map_sizeX != solution_map_sizeY)
+				{
+					ShowWarningError("Error in solution space mapping, suggest adjusting operating constraints.");
+					return -2;
+				}
+				// Check that in this mode the //Outside Air Relative Humidity(0 - 100 % )	//Outside Air Humidity Ratio(g / g)//Outside Air Temperature(�C)
+				if (Mode.MeetsOAEnvConstraints(StepIns.Tosa, Wosa, 100 * StepIns.RHosa)) 
+				{
+					EnvironmentConditionsMet = EnvironmentConditionsMetOnce = true;
+				}
+				else
+				{
+					EnvironmentConditionsMet = false;
+				}
+
+				if (EnvironmentConditionsMet)
+				{
+					Real64 NormalizationReference = GetNormalPoint(1);// assumes the model has at least 1 curve and that all are set the same.
+					if (NormalizationReference == -1)
+					{
+						NormalizationReference = 1;
+						return true;
+					}
+					for (point_number = 0; point_number != solution_map_sizeX; point_number++) // within each mode go though all the combinations of solution spaces.
+					{
+						//Supply Air Mass Flow Rate(kg / s)
+						//Outside Air Fraction(0 - 1)
+						
+						MsaRatio = solutionspace.PointX[point_number];// fractions of rated mass flow rate, so for some modes this might be low but others hi
+						OSAF = solutionspace.PointY[point_number];
+						UnscaledMsa = NormalizationReference * MsaRatio;
+						ScaledMsa = ScaledSystemMaximumSupplyAirMassFlowRate * MsaRatio;
+						Real64 Supply_Air_Ventilation_Volume = 0;
+						//Calculate the ventilation mass flow rate
+						Mvent = ScaledMsa * OSAF;
+
+						if (StdRhoAir > 1)
+						{
+							Supply_Air_Ventilation_Volume = Mvent / StdRhoAir;
+						}
+						else
+						{
+							Supply_Air_Ventilation_Volume = Mvent / 1.225;  //stored as volumetric flow for reporting
+						}
+
+						if (Mvent > MinOA_Msa)
+						{
+							MinVRMet = MinVRMetOnce = true;
+						}
+						else
+						{
+							MinVRMet = false;
+						}
+
+						if (MinVRMet)
+						{
+							//all these points meet the minimum VR requirement
+							solutionspace.PointMeta[point_number] = 1;
+							// Calculate prospective supply air temperature
+							Tsa = Mode.CalculateCurveVal( StepIns.Tosa, Wosa, StepIns.Tra, Wra, UnscaledMsa, OSAF, TEMP_CURVE);
+							// Check it meets constraints
+							if (MeetsSupplyAirTOC(Tsa))
+							{
+								SAT_OC_Met = SAT_OC_MetOnce = SAT_OC_MetinMode = true;
+							}
+							else
+							{
+								SAT_OC_Met = false;
+							}
+							// Calculate prospective supply air Humidity Ratio
+							Wsa = Mode.CalculateCurveVal( StepIns.Tosa, Wosa, StepIns.Tra, Wra, UnscaledMsa, OSAF, W_CURVE);
+							//Return Air Relative Humidity(0 - 100 % ) //Return Air Humidity Ratio(g / g)
+							if (MeetsSupplyAirRHOC(Wsa))
+							{
+				
```

Comment:
@spencerdutton Why does this have to be a shared pointer?  What are you sharing across the multiple references to the same memory?  Unless you have a solid reason for doing it, please make the following changes to avoid this being a red flag:

- Change the definition to simply a list of CSettings instead of a list of pointers
- In this section, just instantiate a new CSettings object and assign the values
- Change any code that attempts to dereference the Settings variable to use standard object dot notation

Then we can move fully away from any pointer concerns with the team.

Reply:
done